simulator/src/subcircuit.js

```diff
@@ -151,6 +151,8 @@ export default class SubCircuit extends CircuitElement {
             this.version = this.savedData.version || "1.0";
 
             this.id = this.savedData.id;
+            this.label = this.savedData.label;
+            this.labelDirection = this.savedData.labelDirection;
```

Comment:
```javascript
this.label = this.savedData.label || "";
this.labelDirection = this.savedData.labelDirection || "RIGHT";
```
This will ensure backward compatibility for subcircuits made before this change

Reply:
Thanks for catching the issue!

Reply:
Thanks, right! Amended and pushed